development/external-plugins/test-untrusted/cmd/main.go

```diff
@@ -39,19 +38,34 @@ func EventHandler(server *externalplugin.Plugin, event externalplugin.Event) {
 		"pr-sender", pr.Sender.Login,
 	)
 	switch pr.Action {
-	case github.PullRequestActionOpened:
+	case github.PullRequestActionOpened, github.PullRequestActionReopened, github.PullRequestActionSynchronize, github.PullRequestActionEdited:
 		pr.GUID = event.EventGUID
 		if pr.Sender.Login == "dependabot[bot]" || pr.Sender.Login == "neighbors-dev-bot" {
+			// If pr action is edited, we check if pr base ref changed. If that's true pr must be retested again.
+			if pr.Action == github.PullRequestActionEdited {
```

Comment:
I was following information from this comment. https://github.com/kubernetes/test-infra/blob/d1f247f5d7711e2f77af97fc5fdc91dce5928319/prow/plugins/trigger/pull-request.go#L95

What I understand this is needed when someone hit edit button on github and change base branch where PR should be merged in.

I did run test and we got only event for action edited.
```
2022-01-21T21:09:10.473Z	info	Received pull request edited action.	{"event-type": "pull_request", "event-GUID": "600f2590-7afe-11ec-8e26-f36737cd5e44", "pr-number": 4768, "pr-sender": "neighbors-dev-bot"}
2022-01-21T21:09:10.473Z	info	PR base not changed, ignoring.	{"event-type": "pull_request", "event-GUID": "600f2590-7afe-11ec-8e26-f36737cd5e44", "pr-number": 4768, "pr-sender": "neighbors-dev-bot"}
```
There is a typo in a condition which prevented sending `/test all` command, but it seems we need pay attention to this event too.

Reply:
But the dependabot will not change the base branch. At best it will close the PR and make a new one with the updated changes, as seen [here](https://github.com/kyma-project/test-infra/pull/4756). I really see no sense in checking that.

Reply:
This plugin can be used for others PRs. Not only for dependabot. This works perfectly fine, doesn't cause performance issues. I don't see much sense in removing it and limiting plugin functionality. I will not remove this.